frontend/pages/hosts/ManageHostsPage/ManageHostsPage.tsx

```diff
@@ -1280,6 +1280,28 @@ const ManageHostsPage = ({
   ) => {
     evt.preventDefault();
 
+    const hiddenColumnsStorage = localStorage.getItem("hostHiddenColumns");
+    let currentHiddenColumns;
+    let visibleColumns;
+    if (hiddenColumnsStorage) {
+      currentHiddenColumns = JSON.parse(hiddenColumnsStorage);
+    }
+
+    if (config && currentUser) {
+      const tableColumns = generateVisibleTableColumns(
+        currentHiddenColumns,
+        config,
+        currentUser,
+        currentTeam
+      );
+
+      const columnAccessors = tableColumns.map((column) =>
+        column.accessor ? column.accessor : ""
+      );
+      columnAccessors.shift();
+      visibleColumns = columnAccessors.join(",");
+    }
+
```

Comment:
This isn't the most elegant solution. A more concise stateful solution was throwing infinite rerender errors. I think fixing that will require refactoring more of the page than we want to do right now, so I'm opting for this approach.

Reply:
Yeah, this page is in pretty bad need of some refactoring TLC